salt/_modules/metalk8s_volumes.py

```diff
@@ -260,6 +307,29 @@ def format(self, force=False):
         # We format a "normal" file, not a block device: we need force=True.
         super(SparseLoopDevice, self).format(force=True)
 
+    @property
+    def is_cleaned_up(self):
+        return not (self.is_provisioned or self.exists)
+
+    def clean_up(self):
+        try:
+            device_path = '/dev/disk/by-uuid/{}'.format(
+                self.get('metadata.uid')
+            )
+            command = ' '.join(['losetup', '--detach', device_path])
+            _run_cmd(command)
+        except CommandExecutionError as exn:
+            # Gracefully handle if the loop device is already detached.
+            # Return code is too generic, have to check the error message…
+            if 'No such file or directory' not in exn.message:
```

Comment:
Not a big fan of this…

Maybe I should use `os.open` + `ioctl(fd, LOOP_CLR_FD, 0)`, error handling would be more fine-grained.